source/cpu/cpu.go

```diff
@@ -263,9 +263,42 @@ func getCPUModel() map[string]string {
 	cpuModelInfo["family"] = strconv.Itoa(cpuid.CPU.Family)
 	cpuModelInfo["id"] = strconv.Itoa(cpuid.CPU.Model)
 
+	hypervisor, err := getHypervisor()
+	if err != nil {
+		klog.ErrorS(err, "failed to detect hypervisor")
+	} else if hypervisor != "" {
+		cpuModelInfo["hypervisor"] = hypervisor
+	}
+
 	return cpuModelInfo
 }
 
+// getHypervisor detects the hypervisor on s390x by reading /proc/sysinfo.
+// If the file does not exist, it returns an empty string with no error.
+func getHypervisor() (string, error) {
+	if _, err := os.Stat("/proc/sysinfo"); os.IsNotExist(err) {
+		return "", nil
+	}
+
+	data, err := os.ReadFile("/proc/sysinfo")
+	if err != nil {
+		return "", err
+	}
+
+	hypervisor := "PR/SM"
```

Comment:
By default we return "PR/SM" even if the machine is not a VM(?)

Reply:
The PR/SM is always the underlying hypervisor, and this is what we expect to see as the default.
This is also consistent to other tooling like `lscpu`.

Reply:
Sorry for my ignorance, but trying to educate myself. On Z you're always having a hypervisor, never running os on "bare metal"?

Reply:
Hi @marquiz, 

The underlying hypervisor that partitions the mainframe resources into logical partitions (LPARs) is always PR/SM. Depending on the use case, in an LPAR either
- a Linux guest is running directly (i.e. Hypervisor is PR/SM),
- or a second hypervisor (KVM or z/VM) is running in the LPAR and partitions the LPAR resources further. The nodes then run as guests in KVM or z/VM.

Thanks to @werled for s390x detailed knowledge.

Reply:
Got that. Thank you @routerhan for educating me.